packages/design-system/src/components/web-components/ds-choice-list/ds-choice-list.tsx

```diff
@@ -0,0 +1,104 @@
+import type * as React from 'react';
+import { ReactNode } from 'react';
+import { define } from '../preactement/define';
+import { ChoiceList, ChoiceListProps } from '../../ChoiceList/ChoiceList';
+import { parseBooleanAttr } from '../wrapperUtils';
+import { ChoiceProps } from '../../ChoiceList/Choice';
+import { findElementsOfType } from '../../utilities/findElementsOfType';
+
+const attributes = [
+  'choices',
+  'class-name',
+  'disabled',
+  'error-id',
+  'error-message-class-name',
+  'error-message',
+  'error-placement',
+  'hint-id',
+  'hint',
+  'inversed',
+  'label-class-name',
+  'label-id',
+  'label',
+  'name',
+  'requirement-label',
+  'root-id',
+  'size',
+  'type',
+] as const;
+
+/* eslint-disable @typescript-eslint/no-namespace */
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-choice-list': React.DetailedHTMLProps<
+        React.HTMLAttributes<HTMLElement> & {
+          [K in (typeof attributes)[number]]?: string;
+        },
+        HTMLElement
+      >;
+    }
+  }
+}
+/* eslint-enable */
+
+interface WrapperProps extends Omit<ChoiceListProps, 'choices' | 'disabled' | 'inversed'> {
+  choices?: string | ChoiceListProps['choices'];
+  disabled?: string;
+  inversed?: string;
+  rootId?: string;
+}
+
+const Wrapper = ({ children, choices, rootId, ...otherProps }: WrapperProps) => {
+  function parseChildren(node: ReactNode): Array<ChoiceProps> {
+    const elements = findElementsOfType(['ds-choice'], node);
+    if (elements.length) {
+      return Array.from(elements).map((element) => {
+        const { children, ...attrs } = element.props;
+
+        /**
+         * Begin logic to define optional nested children of a choice.
+         */
+        let checkedChild,
+          uncheckedChild = undefined;
+
+        if (element.props.children.length > 0) {
+          element.props.children.map((child) => {
+            const { children, slot } = child.props;
```

Comment:
This hasn't been fixed yet. A type of `string | React.ReactElement` will reveal the error, which needs a code solution to be resolved.

Reply:
I'm sorry, I don't understand what the bug is here

Reply:
If you add it to the Astro example with slots in them, it will crash. Some of the `children` in those `ds-choice` children happen to be whitespace characters, which are just strings. So when line 67 tries to access `child.props`, it will get undefined, and then when it tries to pull `{children, slot}` out of it, it will crash because it's trying to access properties of undefined.

Reply:
Adding a type annotation for the `child` parameter on line 66 will more accurately reflect what possible values are being passed to that function, and TypeScript will know that you can't access `props` from something that is possibly just a string.